src/components/SingleDatePicker.jsx

```diff
@@ -84,23 +90,46 @@ const defaultProps = {
 export default class SingleDatePicker extends React.Component {
   constructor(props) {
     super(props);
+
+    this.isTouchDevice = false;
+    this.today = moment();
+
+    this.modifiers = {
+      today: day => this.isToday(day),
+      blocked: day => this.isBlocked(day),
+      // TODO: update these if they change
+      'blocked-calendar': day => props.isDayBlocked(day),
+      'blocked-out-of-range': day => props.isOutsideRange(day),
+      'highlighted-calendar': day => props.isDayHighlighted(day),
+      valid: day => !this.isBlocked(day),
+      hovered: day => this.isHovered(day),
+      selected: day => this.isSelected(day),
+    };
+
+    const initialVisibleMonthThunk = props.initialVisibleMonth || (() => (props.date || moment()));
+    const evaluatedMonth = initialVisibleMonthThunk();
+    const currentMonth = moment.isMoment(evaluatedMonth) ? evaluatedMonth : moment();
+    const visibleDays = getVisibleDays(currentMonth, props.numberOfMonths, props.enableOutsideDays);
```

Comment:
also, all this prop-dependent logic needs to be in both the constructor and `componentWillReceiveProps`. Rather than duplicating it, is there some chunk of it that could be in a shared function?